src/middleware/subscribeWithSelector.ts

```diff
@@ -1,43 +1,62 @@
 import {
-  EqualityChecker,
-  GetState,
-  SetState,
   State,
+  StateCreator,
   StateListener,
-  StateSelector,
-  StateSliceListener,
   StoreApi,
+  StoreMutatorIdentifier,
   Subscribe,
 } from '../vanilla'
 
-export type StoreApiWithSubscribeWithSelector<T extends State> = StoreApi<T> & {
-  subscribe: Subscribe<T> & {
-    (listener: StateListener<T>): () => void
-    <StateSlice>(
-      selector: StateSelector<T, StateSlice>,
-      listener: StateSliceListener<StateSlice>,
-      options?: {
-        equalityFn?: EqualityChecker<StateSlice>
-        fireImmediately?: boolean
-      }
-    ): () => void
+type SubscribeWithSelector = <
+  T extends State,
+  Mps extends [StoreMutatorIdentifier, unknown][] = [],
+  Mcs extends [StoreMutatorIdentifier, unknown][] = []
+>(
+  initializer: StateCreator<
+    T,
+    [...Mps, ['zustand/subscribeWithSelector', never]],
+    Mcs
+  >
+) => StateCreator<T, Mps, [['zustand/subscribeWithSelector', never], ...Mcs]>
+
+declare module '../vanilla' {
```

Comment:
Ah, I bet this does the trick of overloading the interface.

Reply:
Yep